content/blog/2016/2016-05-19-update-plugin-for-pipeline-tutorial.md

```diff
@@ -0,0 +1,189 @@
+
+With that as an example, I was able to start poking at getting a `gatlingArchive` step to show up in the Snippet Generator.  The first thing that I needed to do was to [update the `gatling-plugin` project's `pom.xml` to depend on a recent enough version of Jenkins, as well as specify dependencies on the appropriate pipeline plugins](https://github.com/cprice404/gatling-plugin/commit/b321192bc635eee529ff70e4795591c4594f3664).
+
+Once that was out of the way, I noticed that the `archive` step had some tests written for it, using what looks to be a pretty awesome test API for pipeline jobs and plugins.  Based on [those `archive` tests](https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/300fe6c02b41f072e50a501cfec3e2f425048446/src/test/java/org/jenkinsci/plugins/workflow/steps/ArtifactArchiverStepTest.java#L26-L44), I added [a skeleton for a test for the `gatlingArchive` step](https://github.com/cprice404/gatling-plugin/commit/ed9df4b54c36cee467b3a82e42cb2111e93f9df5) that I was about to write.
+
+Then, I moved on to [actually creating the step](https://github.com/cprice404/gatling-plugin/commit/3de3485be591c7b750ec2671e74558a79efc4319).  The meat of the code was this:
+
+```java
+public class GatlingArchiverStep extends AbstractStepImpl {
+    @DataBoundConstructor
+    public GatlingArchiverStep() {}
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+        public DescriptorImpl() { super(GatlingArchiverStepExecution.class); }
+
+        @Override
+        public String getFunctionName() {
+            return "gatlingArchive";
+        }
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return "Archive Gatling reports";
+        }
+    }
+}
+```
+
+Note that in that commit I also added a `config.jelly` file.  This is how you define the UI for your step, which will show up in the Snippet Generator.  In the case of this Gatling step there's really not much to configure, so my `config.jelly` is basically empty.
+
+With that (and the rest of the code from that commit) in place, I was able to fire up the development Jenkins server (via `mvn hpi:run`, and note that you need to go into the "Manage Plugins" screen on your development server and install the Pipeline plugin once before any of this will work) and visit the Snippet Generator to see if my step showed up in the dropdown:
+
+![gatlingArchive snippet](images/2016-05-19-update-plugin-for-pipeline-tutorial/10_snippet_generator.png)
+
+GREAT SUCCESS!
+
+This step doesn't actually *do* anything yet, but it's recognized by Jenkins and can be included in your pipeline scripts at that point, so, we're on our way, kids!
```

Comment:
I generally try to keep the tone somewhat informal and familiar when writing things like this, but I am not very familiar with the audience demographics for the JenkinsCI blog, so I would be happy to change this if it doesn't seem appropriate for that audience.